### PR TITLE
Sierra: update diagnostic message, pkg-config, Xcode doc

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

* Tweak the messaging of the diagnostic check, including pointing out the current active developer directory.
* Update the `sqlite` pkg-config file.
* Document 10.12's Xcode since it isn't going to change from 8.0 at this point.

### CODE REVIEW

File: Library/Homebrew/extend/os/mac/diagnostic.rb [hunk 1]
```diff
@@ -63,18 +63,22 @@ def check_for_prerelease_xcode
...
+          #{MacOS::Xcode.update_instructions}
+        EOS
+
         if OS::Mac.prerelease?
-          xcode_select_nudge = <<-EOS.undent
+          current_path = Utils.popen_read("/usr/bin/xcode-select", "-p")
+          message += <<-EOS.undent
             If #{MacOS::Xcode.latest_version} is installed, you may need to:
-              sudo xcode-select --switch /path/to/Xcode-beta.app
+              sudo xcode-select --switch /path/to/valid/Xcode.app
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/os/mac/diagnostic.rb]: I reckon it's worth just making this `/Applications` as that's the default location for copy-pasting and anyone else will figure out it needs to be different.

**Comment by MikeMcQuaid** [comment]
👍

File: Library/Homebrew/extend/os/mac/diagnostic.rb (hunk 1 above)
Comment (DomT4) [inline Library/Homebrew/extend/os/mac/diagnostic.rb]: 👍 Done.

**Comment by MikeMcQuaid** [comment]
:shipit:

**Pull request merged**